src/clims/services/substance.py

```diff
@@ -330,6 +339,32 @@ def parents(self):
     def to_ancestry(self):
         return SubstanceAncestry(self)
 
+    def _save_parents(self):
+        parents = [substance_base._wrapped_version for substance_base in self._cached_parents]
+        self._archetype.parents.add(*parents)
+        self._archetype.depth = max([p.depth for p in self._cached_parents]) + 1
```

Comment:
Interesting! I knew of course that we have a graph but not a tree, but I still felt that depth would be a good name for this, probably because I first thought of this as being a tree rather than a graph. But since this is a graph, we should perhaps call this "max_distance" or "level". We can look into that later as the need arises.

Reply:
Ok!